service/src/test/java/greencity/service/HabitAssignServiceImplTest.java

```diff
@@ -289,10 +289,24 @@ void unenrollHabitThrowWrongIdException() {
     @Test
     void deleteHabitAssign() {
         HabitAssign habitAssign = ModelUtils.getHabitAssign();
-        when(habitAssignRepo.findByUserIdAndHabitId(1L, 1L)).thenReturn(Optional.ofNullable(habitAssign));
+        habitAssign.getHabit().setHabitAssigns(generateHabitAssignList());
+        when(habitAssignRepo.findByUserIdAndHabitId(1L, 1L)).thenReturn(habitAssign.getHabit().getHabitAssigns());
         assert habitAssign != null;
         habitAssignService.deleteHabitAssign(1L, 1L);
-        verify(habitAssignRepo).delete(habitAssign);
+        assert HabitAssignStatus.INPROGRESS != habitAssign.getHabit().getHabitAssigns().get(0).getStatus();
+        verify(habitAssignRepo, times(1)).save(any());
```

Comment:
Can you please verify that we save `habitAssign` with the status `HabitAssignStatus.CANCELLED`

Reply:
And please verify that `userShoppingListItemRepo.deleteByShoppingListItemsByHabitAssignId` was called with expected params